### PR TITLE
Fix `'for' loop initial declarations are only allowed in C99 mode`

### PR DESCRIPTION
**Pull request by izhangzhihao**
Fix `'for' loop initial declarations are only allowed in C99 mode`

### CODE REVIEW

File: tensorflow/lite/c/common.c [hunk 1]
```diff
@@ -119,7 +119,8 @@ void TfLiteSparsityFree(TfLiteSparsity* sparsity) {
   }
 
   if (sparsity->dim_metadata) {
-    for (int i = 0; i < sparsity->dim_metadata_size; i++) {
+    int i;
```
Comment (lintian06) [inline tensorflow/lite/c/common.c]: Can you declare and assign `i`?

int i = 0;
for (; i < sparsity->dim_metadata_size; i++) {

Comment (izhangzhihao) [reply tensorflow/lite/c/common.c]: Applied

**Review by lintian06 — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
What toolchain are you compiling with? All compilers that are supported by TF should not complain on this

**Comment by izhangzhihao** [comment]
I'm using this docker image: `amazonlinux:1`

```
sh-4.2# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-amazon-linux/4.8.5/lto-wrapper
Target: x86_64-amazon-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,fortran,ada,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-amazon-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-amazon-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-amazon-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC)
```

**Comment by mihaimaruseac** [comment]
That compiler toolchain is not one of those we support so we cannot help you if you get in errors there.

Can you try using the TensorFlow supplied docker instead?

**Comment by izhangzhihao** [comment]
But this error is the only one I get from compile. I think it will be good to fix this.

**Comment by mihaimaruseac** [comment]
Sure, as this change has a noop semantic, we can take it.

Was just saying in general, especially since there might be other places where this code pattern happens.

**Comment by izhangzhihao** [comment]
Ah, by the way. Let me tell you why I had to use this env to compile. Because I'm trying to using tflite in AWS Lambda. And the env `amazonlinux:1` only give me GLIBC_2.17, but tflite require GLIC_2.27

**Pull request merged**